hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -207,11 +218,17 @@ public StorageStatistics provide() {
           MAX_TOTAL_TASKS, DEFAULT_MAX_TOTAL_TASKS, 1);
       long keepAliveTime = longOption(conf, KEEPALIVE_TIME,
           DEFAULT_KEEPALIVE_TIME, 0);
-      threadPoolExecutor = BlockingThreadPoolExecutorService.newInstance(
+      uploadThreadPoolExecutor = BlockingThreadPoolExecutorService.newInstance(
           maxThreads,
           maxThreads + totalTasks,
           keepAliveTime, TimeUnit.SECONDS,
-          "s3a-transfer-shared");
+          "s3a-upload-shared");
+
+      copyThreadPoolExecutor = BlockingThreadPoolExecutorService.newInstance(
+          maxThreads,
```

Comment:
COPY is server-side (no data transfer) and is thus generally much less resource-intensive and much quicker than PUT (the smaller your bandwidth to S3, the bigger the difference becomes). So I think the `maxThreads` for the copyThreadpool could be (much) higher than for uploadThreadpool and should thus be configurable separately.